manifests/manage.pp

```diff
@@ -46,7 +53,27 @@
 ) {
   $create_resources.each |$type, $resources| {
     $resources.each |$title, $attributes| {
-      create_resources($type, { $title => $attributes })
+      case $type {
+        'file': {
+          if 'template' in $attributes and 'content' in $attributes {
+            fail("You can not set 'content' and 'template' for file ${title}")
+          }
+          if 'template' in $attributes {
+            $content = epp($attributes['template'])
```

Comment:
Should this respect the file extension? I can also imagine you want to pass `epp` to map to the `epp()` function and `template` to map to the `template()` function.

Other thought: you can provide context as well. This may be needed for built in templates that have parameter validation.
```yaml
stdlib::manage::create_resources
  file:
    '/etc/motd':
      epp:
        template: 'profile/motd.epp'
        context: {}
```